### PR TITLE
RFC-0001 Standard tbDEX API Design (APID) Document v0.1.0

### PR DESCRIPTION
**Pull request by KendallWeihe**
See RFC-0001 for full description.

### CODE REVIEW

File: docs/API_DESIGN.md [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+  - [`get_balances()`](#get_balances)
+  - [`create_exchange()`](#create_exchange)
+  - [`submit_order()`](#submit_order)
+  - [`submit_close()`](#submit_close)
+  - [`get_exchange()`](#get_exchange)
+  - [`get_exchanges()`](#get_exchanges)
+
+> [!WARNING]
+>
+> We need to define `JsonNode`
```
Comment (nitro-neal) [inline docs/API_DESIGN.md:52]: I Will think about this... there must be a idiomatic way

**Review by nitro-neal — APPROVED** [review]
Nice 🔥

File: docs/API_DESIGN.md (hunk 1 above)
Comment (KendallWeihe) [reply docs/API_DESIGN.md:52]: Right now I'm referencing the [Custom DSL over in `web5-rs`](https://github.com/TBD54566975/web5-rs/blob/main/docs/CUSTOM_DSL.md), even though we're inside `tbdex-rs`. Basically I think we just need to add the concept of a _JSON Object_ type to the [Custom DSL](https://github.com/TBD54566975/web5-rs/blob/main/docs/CUSTOM_DSL.md) and then update this APID and [the `web5-rs` APID](https://github.com/TBD54566975/web5-rs/blob/main/docs/API_DESIGN.md) to use that new syntax.

File: docs/API_DESIGN.md [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+
+> [!WARNING]
+>
+> Snake vs camel casing is inconsistent
```
[the remark this answers is not in the document]
Comment (KendallWeihe) [reply docs/API_DESIGN.md:56]: just cause move fast

**Pull request merged**